weave

```diff
@@ -1590,23 +1621,30 @@ launch_router() {
         fi
     fi
 
+    # Generate a hash of the password so we can tell if it has changed.
+    # Salt the hash with the peer name to make it even harder to reverse
+    [ -n "$WEAVE_PASSWORD" ] && PASSWORD_HASH="--password-hash=$(echo $PEERNAME$WEAVE_PASSWORD | sha256sum | cut -c1-32)"
```

Comment:
The truncation was intentional; some balance between making the command-line unreadable and the probability of accidental clash.  We even discussed making it shorter to make it harder to reverse.  (There is no 'birthday problem' here)